### PR TITLE
Update Player.cpp Fix Item not deleting after quest

### PR DESCRIPTION
**Pull request by Ashland**
fix item not deleting after quest completion

## CHANGES PROPOSED:
-  
-  

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2620

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Stoabrogga** [comment]
As a side note: This was changed in this commit:
https://github.com/azerothcore/azerothcore-wotlk/commit/999d588c37ef0e56d9043f1aa33ad5e9796f74ca

Perhaps we should just remove the if statement as exactly the same code is executed in the else block. Or we take over the TC implementation:
https://github.com/TrinityCore/TrinityCore/blob/ed939325374216d6a049e778d6ad864c65283ae6/src/server/game/Entities/Player/Player.cpp#L15077-L15080

https://github.com/TrinityCore/TrinityCore/blob/ed939325374216d6a049e778d6ad864c65283ae6/src/server/game/Entities/Player/Player.cpp#L15087-L15090

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -15840,6 +15840,8 @@ void Player::RewardQuest(Quest const* quest, uint32 reward, Object* questGiver,
         {
             if (quest->RequiredItemCount[i] > 0 && itemTemplate->Bonding == BIND_QUEST_ITEM && !quest->IsRepeatable() && !HasQuestForItem(quest->RequiredItemId[i], quest_id, true))
                 DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
+            else
+                DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.cpp]: then why not removing the `if` condition?

Comment (Ashland) [reply src/server/game/Entities/Player/Player.cpp]: Ok I’ll try that and test it

**Comment by Kaev** [comment]
Use this PR instead: https://github.com/azerothcore/azerothcore-wotlk/pull/2626

**Comment by Ashland** [comment]
thank you @Kaev

**Pull request closed without merging**